internal/http/server.go

```diff
@@ -64,20 +64,19 @@ func (srv *Server) GetRequestCount() int {
 }
 
 // Start - start HTTP server
-func (srv *Server) Start() (err error) {
+func (srv *Server) Start(ctx context.Context) (err error) {
 	// Take a copy of server fields.
 	var tlsConfig *tls.Config
 	if srv.TLSConfig != nil {
 		tlsConfig = srv.TLSConfig.Clone()
 	}
 	handler := srv.Handler // if srv.Handler holds non-synced state -> possible data race
 
-	addrs := set.CreateStringSet(srv.Addrs...).ToSlice() // copy and remove duplicates
-
 	// Create new HTTP listener.
 	var listener *httpListener
 	listener, err = newHTTPListener(
-		addrs,
+		ctx,
+		srv.Addrs,
```

Comment:
Any particular reason the dedupe step was removed?

Reply:
So that to add more listeners in future. No need to dedupe since it's already unique.